tests/admin/custom_views/api_key_view_test.py

```diff
@@ -0,0 +1,58 @@
+from unittest.mock import patch
+
+import pcapi.core.offers.factories as offers_factories
+import pcapi.core.users.factories as users_factories
+from pcapi.models import ApiKey
+
+from tests.conftest import TestClient
+from tests.conftest import clean_database
+
+
+class ApiKeyViewTest:
+    @clean_database
+    @patch("wtforms.csrf.session.SessionCSRF.validate_csrf_token")
+    def test_api_key_creation(self, mocked_validate_csrf_token, app):
+        users_factories.UserFactory(email="admin@example.com", isAdmin=True)
+        offerer = offers_factories.OffererFactory(siren=123456789)
+
+        data = dict(
+            offererSiren=offerer.siren,
+        )
+
+        client = TestClient(app.test_client()).with_auth("admin@example.com")
+        response = client.post("/pc/back-office/apikey/new", form=data)
+
+        assert response.status_code == 302
+
+        api_key_create = ApiKey.query.filter_by(offererId=offerer.id).one()
+        assert api_key_create.value is not None
```

Comment:
détail : le suffixe `_create` n'apporte pas grand chose à la compréhension, à mon avis.

```suggestion
        api_key = ApiKey.query.filter_by(offererId=offerer.id).one()
        assert api_key.value is not None
```

Reply:
@dbaty le mieux ça serait que le propriétaire de l'application client puisse généré la clé de son app lui même et supprimer cela de flask admin. Comme tu dis, flask admin n'est pas fait pour ça, ici c'est plutôt un dev app pro. On supprimera ça plus tard comme ça les POs ne sont pas bloqués.

Reply:
OK